pkg/kubelet/util/cache/object_cache_test.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cache
+
+import (
+	"fmt"
+	"testing"
+	"time"
+)
+
+type testObject struct {
+	key string
+	val string
+}
+
+func TestAddAndGet(t *testing.T) {
+	testObj := testObject{
+		key: "foo",
+		val: "bar",
+	}
+	objectCache := NewObjectCache(func() (interface{}, error) {
+		return nil, fmt.Errorf("Unexpected Error: updater should never be called in this test!")
+	}, 1*time.Hour)
+
+	err := objectCache.Add(testObj.key, testObj.val)
+	if err != nil {
+		t.Errorf("Unable to add obj %#v by key: %s", testObj, testObj.key)
+	}
+	value, err := objectCache.Get(testObj.key)
+	if err != nil {
+		t.Errorf("Unable to get obj %#v by key: %s", testObj, testObj.key)
+	}
+	if value.(string) != testObj.val {
+		t.Errorf("Expected to get %#v, but got: %s", testObj, value.(string))
+	}
+
+}
+
+func TestExpirationBasic(t *testing.T) {
+	unexpectedVal := "bar"
+	expectedVal := "bar2"
+
+	testObj := testObject{
+		key: "foo",
+		val: unexpectedVal,
+	}
+	objectCache := NewObjectCache(func() (interface{}, error) {
+		return expectedVal, nil
+	}, 1*time.Second)
```

Comment:
Although 2 seconds is short enough, it would still be better to use fake clock in unit test.
https://github.com/kubernetes/kubernetes/blob/master/pkg/client/cache/expiration_cache_fakes.go